### PR TITLE
fix(nearby view): add more checks for route deduplication

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
The route viewer was getting confused by 2 different routes with the same headsign! What a weird edge case.

### CODE REVIEW

**Review by josh-willis-arcadis — APPROVED** [review]
LGTM!

File: lib/components/viewers/nearby/stop.tsx [hunk 1]
```diff
@@ -38,9 +38,19 @@ const Stop = ({
   const patternRows = (stopData.stoptimesForPatterns || [])
     ?.reduce<PatternStopTime[]>((acc, cur) => {
       const currentHeadsign = extractHeadsignFromPattern(cur.pattern)
-      const dupe = acc.findIndex(
-        (p) => extractHeadsignFromPattern(p.pattern) === currentHeadsign
-      )
+      const dupe = acc.findIndex((p) => {
+        // TODO: use OTP_generated ids
```
Comment (daniel-heppner-ibigroup) [inline lib/components/viewers/nearby/stop.tsx:42]: what do we need to do this?

**Review by daniel-heppner-ibigroup — APPROVED** [review]
lgtm

File: lib/components/viewers/nearby/stop.tsx (hunk 1 above)
Comment (daniel-heppner-ibigroup) [reply lib/components/viewers/nearby/stop.tsx:42]: can we do that? Make a ticket for it and assign it to me if you want

**Pull request merged**